crates/astria-bridge-withdrawer/src/withdrawer/submitter/tests.rs

```diff
@@ -0,0 +1,515 @@
+        startup_guard.wait_until_satisfied(),
+    )
+    .await
+    .unwrap();
+
+    Ok(())
+}
+
+fn make_ics20_withdrawal_action() -> Action {
+    let denom = Denom::from("transfer/channel-0/utia".to_string());
+    let destination_chain_address = "address".to_string();
+    let inner = Ics20Withdrawal {
+        denom: denom.clone(),
+        destination_chain_address,
+        return_address: [0u8; 20].into(),
+        amount: 99,
+        memo: serde_json::to_string(&Ics20WithdrawalMemo {
+            memo: "hello".to_string(),
+            block_number: 1.into(),
+            transaction_hash: [2u8; 32].into(),
+        })
+        .unwrap(),
+        fee_asset_id: denom.id(),
+        timeout_height: IbcHeight::new(u64::MAX, u64::MAX).unwrap(),
+        timeout_time: 0, // zero this for testing
+        source_channel: "channel-0".parse().unwrap(),
+    };
+
+    Action::Ics20Withdrawal(inner)
+}
+
+fn make_bridge_unlock_action() -> Action {
+    let denom = Denom::from("nria".to_string());
+    let inner = BridgeUnlockAction {
+        to: [0u8; 20].into(),
+        amount: 99,
+        memo: serde_json::to_vec(&BridgeUnlockMemo {
+            block_number: 1.into(),
+            transaction_hash: [2u8; 32].into(),
+        })
+        .unwrap(),
+        fee_asset_id: denom.id(),
+    };
+    Action::BridgeUnlock(inner)
+}
+
+fn make_batch_with_bridge_unlock_and_ics20_withdrawal() -> Batch {
+    Batch {
+        actions: vec![make_ics20_withdrawal_action(), make_bridge_unlock_action()],
+        rollup_height: 10,
+    }
+}
+
+fn make_tx_commit_success_response() -> tx_commit::Response {
+    tx_commit::Response {
+        check_tx: CheckTx::default(),
+        tx_result: ExecTxResult::default(),
+        hash: vec![0u8; 32].try_into().unwrap(),
+        height: Height::default(),
+    }
+}
+
+fn make_tx_commit_check_tx_failure_response() -> tx_commit::Response {
+    tx_commit::Response {
+        check_tx: CheckTx {
+            code: 1.into(),
+            ..CheckTx::default()
+        },
+        tx_result: ExecTxResult::default(),
+        hash: vec![0u8; 32].try_into().unwrap(),
+        height: Height::default(),
+    }
+}
+
+fn make_tx_commit_deliver_tx_failure_response() -> tx_commit::Response {
+    tx_commit::Response {
+        check_tx: CheckTx::default(),
+        tx_result: ExecTxResult {
+            code: 1.into(),
+            ..ExecTxResult::default()
+        },
+        hash: vec![0u8; 32].try_into().unwrap(),
+        height: Height::default(),
+    }
+}
+
+/// Convert a `Request` object to a `SignedTransaction`
+fn signed_tx_from_request(request: &Request) -> SignedTransaction {
+    use astria_core::generated::protocol::transaction::v1alpha1::SignedTransaction as RawSignedTransaction;
+    use prost::Message as _;
+
+    let wrapped_tx_sync_req: request::Wrapper<tx_sync::Request> =
+        serde_json::from_slice(&request.body)
+            .expect("can't deserialize to JSONRPC wrapped tx_sync::Request");
```

Comment:
```suggestion
            .expect("able to deserialize to JSONRPC wrapped tx_sync::Request");
```

`expect` language is the opposite of an error message, you want to put what you expected to happen (but didn't, if the expect fails)